util_test.go

```diff
@@ -0,0 +1,25 @@
+package shapefile
+
+import (
+	"github.com/alecthomas/assert/v2"
+	"testing"
+)
+
+func TestIsMACOSX(t *testing.T) {
+	type testCase struct {
+		path     string
+		expected bool
+	}
+	testCases := []testCase{
+		{"__MACOSX/dir/._test.shp", true},
+		{"dir/__MACOSX/._test.shp", true},
+		{"dir/__MACOSX/dir/._test.shp", true},
+		{"dir/__MACOSX/dir/__MACOSX/._test.shp", true},
+		{"dir/._test.shp", false},
```

Comment:
Please add testcases for:

```go
{"dir/ABC__MACOSX/._test.shp", false}.
{"dir/__MACOSXABC/._test.shp", false}.
{"dir/._test.shp.__MACOSX", false}.
```

Hint: you will need to modify the code to use `strings.Split` with `path.Separator`.